src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -346,4 +342,39 @@ public function get_collection_params() {
 		 */
 		return apply_filters( 'rest_pattern_directory_collection_params', $query_params );
 	}
+
+	/*
+	 * Include a hash of the query args, so that different requests are stored in
+	 * separate caches.
+	 *
+	 * MD5 is chosen for its speed, low-collision rate, universal availability, and to stay
+	 * under the character limit for `_site_transient_timeout_{...}` keys.
+	 *
+	 * @link https://stackoverflow.com/questions/3665247/fastest-hash-for-non-cryptographic-uses
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $query_args Query arguments to generate a transient key from.
+	 *
+	 * @return string Transient key.
+	 */
+	protected function get_transient_key( $query_args ) {
+
+		if ( isset( $query_args['slug'] ) ) {
+			// This is an additional precaution because the "sort" function expects an array.
+			$query_args['slug'] = (array) $query_args['slug'];
+
+			// Sort the array so that the transient key doesn't depend on the order of slugs.
+			sort( $query_args['slug'] );
+
+			// Slugs have to be imploded separately as implode doesn't work with recursive arrays.
+			$query_args['slug'] = implode( ',', $query_args['slug'] );
+
+			if ( '' === trim( $query_args['slug'] ) ) {
```

Comment:
```suggestion
			$query_args['slug'] = trim( implode( ',', $query_args['slug'] ) );

			if ( '' === $query_args['slug'] ) {
```

Reply:
I think code is not needed anymore because of 2c08eeab216148827b02aa5f99a309948ca31a3b
`serialize` supports multidimensional arrays and we don't have to implode values.